src/AppBundle/TonMacron/TonMacronSerializer.php

```diff
@@ -0,0 +1,97 @@
+<?php
+
+namespace AppBundle\TonMacron;
+
+use AppBundle\Entity\TonMacronChoice;
+use AppBundle\Entity\TonMacronFriendInvitation;
+
+class TonMacronSerializer
+{
+    /**
+     * @param TonMacronChoice[] $choices
+     *
+     * @return string
+     */
+    public function serializeChoices(array $choices): string
+    {
+        if (!is_iterable($choices)) {
+            throw new \InvalidArgumentException();
+        }
+
+        $handle = fopen('php://memory', 'rb+');
```

Comment:
I won't do this right now as I have a lot to do, but feel free to improve this tomorrow :) .